scipy/stats/_qmc.py

```diff
@@ -914,13 +910,48 @@ class LatinHypercube(QMCEngine):
 
         .. versionadded:: 1.8.0
 
+    strength : {1, 2}, optional
+        Strength of the LHS. ``strength=1`` produces a plain LHS while
+        ``strength=2`` produces an orthogonal array based LHS of strength 2
+        [7]_, [8]_. In that case, only ``n=p**2`` points can be sampled,
+        with ``p`` a prime number. It also constrains ``d <= p + 1``.
+        Default is 1.
+
+        .. versionadded:: 1.8.0
+
     seed : {None, int, `numpy.random.Generator`}, optional
         If `seed` is None the `numpy.random.Generator` singleton is used.
         If `seed` is an int, a new ``Generator`` instance is used,
         seeded with `seed`.
         If `seed` is already a ``Generator`` instance then that instance is
         used.
 
+    Notes
+    -----
+
+    When LHS is used for integrating a function :math:`f` over :math:`n`,
+    LHS is extremely effective on integrands that are nearly additive [2]_.
+    With a LHS of :math:`n` points, the variance of the integral is always
+    lower than plain MC on :math:`n-1` points [3]_. There is a central limit
+    theorem for LHS on the mean and variance of the integral [4]_, but not
+    necessarily for optimized LHS due to the randomization.
+
+    :math:`A` is called an orthogonal array of strength :math:`t` if in each
+    n-row-by-t-column submatrix of :math:`A`: all :math:`p^t` possible
+    distinct rows occur the same number of times,
+    with :math:`p` a prime number. The elements of :math:`A` are in the set
```

Comment:
```suggestion
    distinct rows occur the same number of times.
    The elements of :math:`A` are in the set
```
Art said `If it is not a prime then you cannot use modular arithmetic.` 
So it still seems possible to me to have a strength 2 array without `sqrt(n)` being prime.

Reply:
He is further adding

> You should raise an error if they ask for something that cannot be generated.  For modular arithmetic that would be asking for a non prime.  If you connect up the code to some Galois Field code then you could accept prime powers for the number of levels.  Just be sure that no algorithm uses modular arithmetic outside of the prime setting.
> ...
> You should not raise an error for strength 1 and an arbitrary sample size.  It is only the corner case: strength > 1 AND non-prime number of levels that needs to trigger an error.   With strength 2 some sample sizes are grayed out.

I also think this is specific to modular arithmetic. But since I actually don't get this part (in practice what it is, what do you do with the sample), I would prefer to be conservative for now. We can always revert this decision and have a warning instead of raising for instance.

Still, I will add a note about why primes are needed in the doc.

Reply:
This is fine. Because we don't understand, I don't want to write something that can be interpreted as "For an array to be considered strength 2 orthogonal, `p` must be prime." It is fine, though, to state that `p` must be prime to use the function. We just don't have to say anything about whether `p` must be prime in general.

Reply:
ok then let me know what you think about my edits.